test/functional/forall/test-forall-atomic-utils.hpp

```diff
@@ -86,4 +88,73 @@ using AtomicDataTypeList =
               double
            >;
 
+
+// range segment multiplexer
+template< typename Index, typename SegType >
+struct RSMultiplexer {};
+
+template< typename Index >
+struct RSMultiplexer < Index, RAJA::TypedRangeSegment<Index> >
+{
+  RAJA::TypedRangeSegment<Index>
+  makeseg( Index N )
+  {
+    return RAJA::TypedRangeSegment<Index>( 0, N );
+  }
+};
+
+template< typename Index >
+struct RSMultiplexer < Index, RAJA::TypedRangeStrideSegment<Index> >
+{
+  RAJA::TypedRangeStrideSegment<Index>
+  makeseg( Index N )
+  {
+    return RAJA::TypedRangeStrideSegment<Index>( 0, N, 1 );
+  }
+};
+
+template< typename Index >
+struct RSMultiplexer < Index, RAJA::TypedListSegment<Index> >
+{
+  RAJA::TypedListSegment<Index>
+  makeseg( Index N )
+  {
+    std::vector<Index> temp(N);
+    std::iota( std::begin(temp), std::end(temp), 0 );
+    return RAJA::TypedListSegment<Index>( &temp[0], static_cast<size_t>(temp.size()) );
```

Comment:
I don't understand how this works for non-CPU tests. You should have to pass a camp Resource object into the list segment constructor so it can create a copy of the indices in the proper memory space.

Reply:
Did the default behavior change to host instead of managed when built with cuda?
If we only run on our P9 systems with ATS then host memory is device accessible.
I agree that this should use a camp resource so it will work on any gpu system.

Reply:
This has not been tested on a machine like RZMANTA, but I will make this change to all of the atomic forall tests anyway for good measure.

Reply:
Take a look at the forall segment tests to see how Mike and I used the camp resource objects for that. 

@MrBurmark the existing behavior for how list segments manage their index data (compile-time choice based on whether cuda or hip is enabled) is still there until we rip that out after the next release. Then, we will require users to supply camp resources so they can do what they want for any memory resource at runtime.

Reply:
@MrBurmark @rhornung67 Strange, all of the CUDA tests are failing on RZMANTA with the following. Have you seen this before?
[----------] 2 tests from Cuda/ForallIndexSetTest/0, where TypeParam = camp::list<long, camp::resources::v1::Cuda, RAJA::policy::indexset::ExecPolicy<RAJA::policy::sequential::seq_exec, RAJA::policy::cuda::cuda_exec<128ul, false> > >
[ RUN      ] Cuda/ForallIndexSetTest/0.IndexSetForall
CUDAassert: no kernel image is available for execution on the device /g/g20/chen59/work/RAJAatomictests/rajaatomicrangesMANTA/include/RAJA/policy/cuda/MemUtils_CUDA.hpp 210
unknown file: Failure
C++ exception with description "CUDAassert" thrown in the test body.
[  FAILED  ] Cuda/ForallIndexSetTest/0.IndexSetForall, where TypeParam = camp::list<long, camp::resources::v1::Cuda, RAJA::policy::indexset::ExecPolicy<RAJA::policy::sequential::seq_exec, RAJA::policy::cuda::cuda_exec<128ul, false> > > (153 ms)

Reply:
But, it works on rzansel? What 'sm' level are you compiling with?

Reply:
Originally used sm_70, tried sm_60 but got the same error. I'm trying different CUDA versions now.

Reply:
All of the tests pass with the latest CUDA version 10.2.89 (even the tests where I did not use the CAMP resource). I'll add the camp::resources and update the build scripts to use the latest CUDA.

Reply:
That's weird. DId they update the system software on rzmanta? Last time I tried cuda 10.2.*, there was a version mismatch with the cuda runtime.

Reply:
I guess they must have updated 10.2.89, but I haven't seen any official announcement from LC or John Gyllenhaal. The RAJA tests pass on RZANSEL with 10.2.89, so I will update the build scripts accordingly.

Reply:
You mean add build scripts, correct?  We still need to keep the 10.1 scripts around because all our users are using 10.1.x.

Do the tests compile and pass with CUDA 10.1 on both machines?

Reply:
CUDA 10.1 builds and runs correctly on RZANSEL, and builds correctly on RZMANTA but fails the CUDA test runs (seq and omp tests work). I could add 10.2.89 scripts and keep the 10.1's around, or do you think I should try keeping the current set of scripts and altering the CUDA CMAKE options depending on the platform detected?

Reply:
Keep the existing scripts and rename to include "10.1" instead of just "10".  Then add some for "10.2" using that in the name.

Have you contacted LC about what's going on with rzmanta?  It should work for both CUDA versions.